### PR TITLE
CAPT-1415 Changes to SLC in ECP/LUP journey

### PR DESCRIPTION
**Pull request by your**
https://dfedigital.atlassian.net/browse/CAPT-1415

This PR covers:

1. The removal of student loan questions from the combined journey (ECP/LUP)
2. The addition of an automated task for the student loan plan (ECP/LUP only)

The best way to start this review is to understand that at the moment we are determining the student loan plan _after_ the applicant answered a whole bunch of questions, and when the user changes one of these answers; look [here](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2562/files#diff-177255fb498b99ec75c3866174729f24e11d030819b91655a5913b28669505c2L494)

This question-based determination is not needed anymore as we currently hold SLC data for past applicants, and, should we not have that, we can still allow the claim to be submitted and request further info to the SLC.

For TSLR, as per #2529, we always run an automated task "student loan **amount**".
For ECP/LUP, as per current PR, we'll alway run the "student loan **plan**" task instead – unless:
1. we held information on the applicant's student loan repayments at the time of submission
2. the claim was submitted before this automation work is deployed (*)

It's probably worth referring to the Lucid Board linked in [this](https://dfedigital.atlassian.net/browse/CAPT-1415?focusedCommentId=78301) comment to understand these scenarios.

(*) At the moment, the Ops team don't run admin checks on the student loan plan for ECP/LUP, essentially trusting user input. This "extra" new check will be part of their process once this work is live.

Some refactoring was necessary. [This](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2541#discussion_r1446304903) comment has been addressed with [this](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2562/commits/525961b7314cdcb375ab40705431cedcd3780fcd) commit.

### CODE REVIEW

**Review by slorek — APPROVED** [review]
Looks good, just a few suggestions and comments.

File: app/controllers/claims_controller.rb [hunk 1]
```diff
@@ -256,4 +254,11 @@ def bank_account
   def correct_policy_namespace?
     PolicyConfiguration.policies_for_routing_name(params[:policy]).include?(current_claim.policy)
   end
+
+  def retrieve_student_loan_details
+    # student loan details are retrieved every time the user confirms their details
+    return unless ["personal-details", "teacher-detail"].include?(params[:slug])
```
[the remark this answers is not in the document]
Comment (your) [reply app/controllers/claims_controller.rb]: Done with [ad27e98](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2562/commits/ad27e9815919af5b7131d9a647d27faf9072050f)

File: app/controllers/claims_controller.rb [hunk 2]
```diff
@@ -69,6 +66,7 @@ def update
     one_time_password
 
     if current_claim.save(context: page_sequence.current_slug.to_sym)
+      retrieve_student_loan_details
```
[the remark this answers is not in the document]
Comment (your) [reply app/controllers/claims_controller.rb:69]: I'm restricting it with [8c0ca96](https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2562/commits/8c0ca966d616fbf03ad3f05bfbdc39e748660390) for now, but as you say we'll have to redesign how these behaviours are called per policy. I had a peek at the TID branch and it's getting even more complicated indeed the `update` action. This sounds like a ticket for the architecture improvements epic!

File: app/helpers/admin/claims_helper.rb [hunk 3]
```diff
@@ -146,7 +146,7 @@ def task_status_tag(claim, task_name)
       elsif task.claim_verifier_match_none?
         status = "No match"
         status_colour = "red"
-      elsif task.claim_verifier_match.nil? && %w[census_subjects_taught employment induction_confirmation student_loan_amount].include?(task_name)
+      elsif task.claim_verifier_match.nil? && %w[census_subjects_taught employment induction_confirmation student_loan_amount student_loan_plan].include?(task_name)
```
[the remark this answers is not in the document]
Comment (your) [reply app/helpers/admin/claims_helper.rb:149]: Very true, another ticket for the improvements perhaps? I think this particular conditional branch is demonstrating that the list of possible task outcome values (an enum!) are not sufficient as the outcome is quite dependant on the specific task and even policy. I'd go even one step further and argue that the `claim_verifier_match` is not a valid attribute name as some tasks don't do simple matching against a data source but have more complex rules embedded (unfortunately, and it's the case of all the tasks in this list)

File: app/models/automated_checks/claim_verifier.rb [hunk 4]
```diff
@@ -27,6 +27,10 @@ def initialize(
         ClaimVerifiers::StudentLoanAmount.new(
           admin_user: admin_user,
           claim: claim
+        ),
+        ClaimVerifiers::StudentLoanPlan.new(
```
[the remark this answers is not in the document]
Comment (your) [reply app/models/automated_checks/claim_verifier.rb:31]: Ye, thanks for the heads up. I was keeping an eye and definitely something that I will have to deal it when rebasing.

File: app/helpers/admin/claims_helper.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (slorek) [reply app/helpers/admin/claims_helper.rb:149]: Agreed, this attribute seems to have been [mis/ab]used in the past!

[the remark this answers is not in the document]
Comment (your) [reply app/helpers/admin/claims_helper.rb:149]: Here's just another example of how I have to re-use the "none" value for more than one condition (and potential outcome, although it's always NO MATCH) while executing a particular task: https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2591

**Pull request merged**